### PR TITLE
binary_cross_entropy_with_logits

### LINKED ISSUE #6993: binary_cross_entropy_with_logits

### PR DESCRIPTION
**Pull request by bipinKrishnan**
Adds binary cross entropy with logits loss function to the torch frontend.

Closes #6993

### CODE REVIEW

**Comment by bipinKrishnan** [comment]
I am not sure if BCE with logits should be included in the Ivy functional API and then used inside the torch frontend. The reason to include is because, similar to binary cross entropy(which is already implemented in the ivy api), BCE with logits is present in both tensorflow and pytorch. 

But unlike binary cross entropy, the usage is not uniform among the frameworks due to positional weights. Tensorflow uses ```tf.nn.weighted_cross_entropy_with_logits``` when positional weights are present and ```tf.nn.sigmoid_cross_entropy_with_logits``` when they are absent, while pytorch does both of these using ```torch.nn.functional.binary_cross_entropy_with_logits```(positional weights are taken as an optional argument here).

Any feedback regarding this would be great :slightly_smiling_face:.

**Comment by bipinKrishnan** [comment]
@ZoeCD Can we merge this PR or is there any changes that needs to be added?

**Comment by ZoeCD** [comment]
Hi! Other members of the team and I were discussing your question. We think that it's probably good to implement an ivy function for computing the binary cross entropy with logits. An extra argument can define how the logits are computed, so the difference between frameworks can be handled. Feel free to create a separate issue named "ivy.binary_cross_entropy_with_logits" to add it to the next release to-do list. If you wish to work on the new function yourself, we can keep this PR open for now so that we can use the new ivy function in the frontend. Hope this makes sense! 🙂

**Comment by bipinKrishnan** [comment]
Cool! I will raise an issue for ```ivy.binary_cross_entropy_with_logits``` and work on a PR for the same. Once that is merged, I will continue my work here.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by bipinKrishnan** [comment]
Adding this comment so that the PR is not closed automatically by marking as stale.

**Comment by bipinKrishnan** [comment]
@ZoeCD The PR is ready for review now. Let me know if anything needs to be added :)

**Comment by ZiadAmerr** [comment]
Hi, @bipinKrishnan, is this comment visible to you?

**Comment by bipinKrishnan** [comment]
@ZoeCD a friendly ping here.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ZoeCD** [comment]
Hi! So sorry for the late reply 😅. The changes look good to me. However, it seems there is a bug in `ivy.binary_cross_entropy_with_logits` You can look for the details on test-frontend-torch raw logs. I guess we can merge this PR since the changes needed are not from this part of the code. Let me know if you want to merge it or if you prefer to wait until you fix the bugs in the ivy function

**Comment by bipinKrishnan** [comment]
@ZoeCD  I'll go ahead with merging this PR for now.
